Algolia.Search/Index.cs

```diff
@@ -1000,5 +1000,199 @@ public JObject SearchDisjunctiveFaceting(Query query, IEnumerable<string> disjun
+        }
+
+        /// <summary>
+        /// Synchronously call <see cref="Index.BrowseFromAsync"/>.
+        /// </summary>
+        /// <param name="forwardToSlaves">Forward the operation to the slave indices</param>
+        public JObject ClearSynonyms(bool forwardToSlaves = false)
+        {
+            return ClearSynonymsAsync(forwardToSlaves).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Add or Replace a list of synonyms 
+        /// </summary>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        /// <param name="replaceExistingSynonyms">Replace the existing synonyms with this batch</param>
+        public Task<JObject> BatchSynonymsAsync(IEnumerable<object> objects, bool forwardToSlaves = false, bool replaceExistingSynonyms = false, CancellationToken token = default(CancellationToken))
+        {
+            return _client.ExecuteRequest(AlgoliaClient.callType.Write, "POST", string.Format("/1/indexes/{0}/synonyms/batch?replaceExistingSynonyms={1}&forwardToSlaves={2}", _urlIndexName, replaceExistingSynonyms ? "true" : "false", forwardToSlaves ? "true" : "false"), objects, token);
+        }
+
+        /// <summary>
+        /// Synchronously call <see cref="Index.BatchSynonymsAsync"/>.
+        /// </summary>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        /// <param name="replaceExistingSynonyms">Replace the existing synonyms with this batch</param>
+        public JObject BatchSynonyms(IEnumerable<object> objects, bool forwardToSlaves = false, bool replaceExistingSynonyms = false)
+        {
+            return BatchSynonymsAsync(objects, forwardToSlaves, replaceExistingSynonyms).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Update one synonym
+        /// <param name="objectID">The objectID of the synonym</param>
+        /// <param name="content">The new content of this synonym</param>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        /// </summary>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        public Task<JObject> SaveSynonymAsync(string objectID, object content, bool forwardToSlaves = false, CancellationToken token = default(CancellationToken))
+        {
+            return _client.ExecuteRequest(AlgoliaClient.callType.Write, "PUT", string.Format("/1/indexes/{0}/synonyms/{1}?forwardToSlaves={2}", _urlIndexName,  Uri.EscapeDataString(objectID), forwardToSlaves ? "true" : "false"), content, token);
+        }
+
+        /// <summary>
+        /// Synchronously call <see cref="Index.SaveSynonymAsync"/>.
+        /// </summary>
+        /// <param name="forwardToSlave">Forward the operation to the slave indices</param>
+        public JObject SaveSynonym(string objectID, object content, bool forwardToSlaves = false, bool replaceExistingSynonyms = false)

```

Comment:
`replaceExistingSynonyms` is not used at all here, and is not a valid param for the PUT request that is sent for `SaveSynonym`